Exiled.CustomRoles/API/Features/KeypressActivator.cs

```diff
@@ -121,6 +99,34 @@ private IEnumerator<float> ProcessAltKey(Player player)
             altTracker[player] = 0;
         }
 
+        private void OnTogglingNoClip(TogglingNoClipEventArgs ev)
+        {
+            if (ev.Player.IsNoclipPermitted)
+                return;
+
+            if (!ActiveAbility.AllActiveAbilities.ContainsKey(ev.Player))
+                return;
+
+            if (!altTracker.ContainsKey(ev.Player))
+                altTracker.Add(ev.Player, 0);
+
+            altTracker[ev.Player]++;
+
+            if (!coroutineTracker.ContainsKey(ev.Player))
+                coroutineTracker.Add(ev.Player, default);
+
+            if (!coroutineTracker[ev.Player].IsRunning)
+                coroutineTracker[ev.Player] = Timing.RunCoroutine(ProcessAltKey(ev.Player));
+        }
+
+        private void OnEndingRound(EndingRoundEventArgs ev)
+        {
+            altTracker.Clear();
+            foreach (CoroutineHandle handle in coroutineTracker.Values)
+                Timing.KillCoroutines(handle);
```

Comment:
`Timing.KillCoroutines(coroutineTracker.Values.ToArray());`
May be better.

Reply:
you just make me learn this BRUH